src/GafferImage/Warp.cpp

```diff
@@ -355,6 +361,43 @@ void Warp::compute( Gaffer::ValuePlug *output, const Gaffer::Context *context )
 		const Engine *engine = engineData->engine;
 
 
+		// Start by testing if the tile is completely empty
+		// We abort this test on the first valid position returned from the engine, but
+		// worst case, we could traverse the whole tile, only to find one valid pixel in
+		// the last corner we check, and then we will have to revisit the whole tile below.
+		// It's still a definite performance win, because in the common cases we usually
+		// either abort this test quickly, or we are actually in an invalid tile, and noticing
+		// this as soon as possible is a big win.
+		bool emptyTile = true;
+		for( int y = 0; y < ImagePlug::tileSize(); y++ )
+		{
+			for( int x = 0; x < ImagePlug::tileSize(); x++ )
+			{
+				if( BufferAlgo::contains( dataWindow, V2i( tileOrigin.x + x, tileOrigin.y + y ) ) )
+				{
+					V2f inputPosition = engine->inputPixel( V2f(
```

Comment:
indentation seems a bit off here. assuming you want to keep it multiline, either:

```
V2f inputPosition = engine->inputPixel( V2f(
    ( tileOrigin.x + x ) + 0.5,
    ( tileOrigin.y + y ) + 0.5
) );
```

or:

```
V2f inputPosition = engine->inputPixel(
    V2f(
        ( tileOrigin.x + x ) + 0.5,
        ( tileOrigin.y + y ) + 0.5
    )
);
```